report/email.go

```diff
@@ -34,54 +34,94 @@ type EMailWriter struct{}
 
 func (w EMailWriter) Write(rs ...models.ScanResult) (err error) {
 	conf := config.Conf
-	to := strings.Join(conf.EMail.To[:], ", ")
-	cc := strings.Join(conf.EMail.Cc[:], ", ")
-	mailAddresses := append(conf.EMail.To, conf.EMail.Cc...)
-	if _, err := mail.ParseAddressList(strings.Join(mailAddresses[:], ", ")); err != nil {
-		return fmt.Errorf("Failed to parse email addresses: %s", err)
-	}
+	var message string
+	var totalResult models.ScanResult
+	sender := NewEMailSender()
 
 	for _, r := range rs {
-		var subject string
-		if len(r.Errors) != 0 {
-			subject = fmt.Sprintf("%s%s An error occurred while scanning",
-				conf.EMail.SubjectPrefix, r.ServerInfo())
+		if conf.FormatOneEMail {
+			message += toFullPlainText(r) + "\r\n\r\n"
+			totalResult.KnownCves = append(totalResult.KnownCves, r.KnownCves...)
+			totalResult.UnknownCves = append(totalResult.UnknownCves, r.UnknownCves...)
 		} else {
-			subject = fmt.Sprintf("%s%s %s",
-				conf.EMail.SubjectPrefix, r.ServerInfo(), r.CveSummary())
+			var subject string
+			if len(r.Errors) != 0 {
+				subject = fmt.Sprintf("%s%s An error occurred while scanning",
+					conf.EMail.SubjectPrefix, r.ServerInfo())
+			} else {
+				subject = fmt.Sprintf("%s%s %s",
+					conf.EMail.SubjectPrefix, r.ServerInfo(), r.CveSummary())
+			}
+			message = toFullPlainText(r)
+			if err := sender.Send(subject, message); err != nil {
+				return err
+			}
 		}
+	}
 
-		headers := make(map[string]string)
-		headers["From"] = conf.EMail.From
-		headers["To"] = to
-		headers["Cc"] = cc
-		headers["Subject"] = subject
-		headers["Date"] = time.Now().Format(time.RFC1123Z)
-		headers["Content-Type"] = "text/plain; charset=utf-8"
-
-		var message string
-		for k, v := range headers {
-			message += fmt.Sprintf("%s: %s\r\n", k, v)
-		}
-		message += "\r\n" + toFullPlainText(r)
-
-		smtpServer := net.JoinHostPort(conf.EMail.SMTPAddr, conf.EMail.SMTPPort)
-		err = smtp.SendMail(
-			smtpServer,
-			smtp.PlainAuth(
-				"",
-				conf.EMail.User,
-				conf.EMail.Password,
-				conf.EMail.SMTPAddr,
-			),
-			conf.EMail.From,
-			conf.EMail.To,
-			[]byte(message),
+	if conf.FormatOneEMail {
+		subject := fmt.Sprintf("%s %s",
```

Comment:
How do you think it would be better to display a summary at the top of the mail?

```
               message = fmt.Sprintf(
                       `
One Line Summary
================
%s


%s`,
                       toOneLineSummary(rs...), message)
```